homedecor_seating/init.lua

```diff
@@ -132,41 +145,68 @@ function lrfurn.sit(pos, node, clicker, itemstack, pointed_thing, seats)
 	--seat the player
 	clicker:set_pos(sit_pos)
 
-	xcompat.player.player_attached[name] = true
-    xcompat.player.set_animation(clicker, "sit", 0)
-	if has_player_monoids then
-		physics_cache[name] = true
-		player_monoids.speed:add_change(clicker, 0, "homedecor_seating:sit")
-		player_monoids.jump:add_change(clicker, 0, "homedecor_seating:sit")
-		player_monoids.gravity:add_change(clicker, 0, "homedecor_seating:sit")
+	--we only care about 4 rotations, but just in case someone worldedits, etc - do something other than crash
+	--radians are stupid, using degrees and then converting
+	local p2r = {
+		0*math.pi/180,
+		0*math.pi/180, --correct
+		180*math.pi/180, --correct
+		90*math.pi/180, --correct
+		270*math.pi/180, --correct
+		0*math.pi/180,
+		0*math.pi/180,
+		0*math.pi/180,
+	}
+	p2r[0] = p2r[1]
+
+	local p2r_sofa = {
+		0*math.pi/180,
+		90*math.pi/180, --correct
+		270*math.pi/180, --correct
+		180*math.pi/180, --correct
+		0*math.pi/180, --correct
+		0*math.pi/180,
+		0*math.pi/180,
+		0*math.pi/180,
+	}
+	p2r_sofa[0] = p2r_sofa[1]
+
+	local p2r_facedir = {
+		[0] = 180*math.pi/180,
+		[1] = 90*math.pi/180,
+		[2] = 0*math.pi/180,
+		[3] = 270*math.pi/180,
+	}
```

Comment:
Can you move this out of the function scope to reduce clutter and save some cpu cycles?

Reply:
https://github.com/mt-mods/homedecor_modpack/pull/81/commits/39761fc9c11f20ad30566343fc653c9322469b73